core/src/databases/table.rs

```diff
@@ -356,11 +376,11 @@ impl LocalTable {
 
     pub async fn upsert_rows(
         &self,
-        store: Box<dyn Store + Sync + Send>,
-        databases_store: Box<dyn DatabasesStore + Sync + Send>,
+        store: &Box<dyn Store + Sync + Send>,
```

Comment:
To be consistent would love to keep the Box<...>. Box is a pointer so it's virtually free to clone. Any reason to move to a reference here?